Description: Disable assert macro redefinition in CEF cef_logging.h for abseil/perfetto
 CEF's cef_logging.h (line 670-672) redefines assert to DLOG_ASSERT. This
 breaks abseil's str_format constexpr code and perfetto's
 scattered_stream_writer.h which require the standard assert macro from <cassert>.
 .
 When is_official_build=true, these compile-time checks are enabled and any
 code path that includes cef_logging.h before these third-party headers will
 fail with "use of undeclared identifier 'assert'" errors.
 .
 This is the CEF companion patch to 0042 and 0043 Chromium patches, which fix
 the same issue in base/logging.h and partition_alloc/logging.h.
Author: Juan Manuel Mendez Rey <vejeta@gmail.com>
Bug-Debian: https://bugs.debian.org/915400
Forwarded: not-needed
Last-Update: 2025-12-05

diff --git a/include/base/cef_logging.h b/include/base/cef_logging.h
index 9ca4ae1..cce45d1 100644
--- a/include/base/cef_logging.h
+++ b/include/base/cef_logging.h
@@ -667,9 +667,11 @@ const LogSeverity LOG_DCHECK = LOG_INFO;

 #define NOTREACHED() DCHECK(false)

-// Redefine the standard assert to use our nice log files
-#undef assert
-#define assert(x) DLOG_ASSERT(x)
+// CEF originally redefined assert to DLOG_ASSERT, but this breaks abseil's
+// str_format and perfetto's constexpr code which require the standard assert
+// macro from <cassert>. Disable the redefinition for is_official_build=true.
+// #undef assert
+// #define assert(x) DLOG_ASSERT(x)

 // This class more or less represents a particular log message.  You
 // create an instance of LogMessage and then stream stuff to it.
